packages/serve-runtime/src/createServeRuntime.ts

```diff
@@ -105,18 +106,60 @@ export function createServeRuntime<TContext extends Record<string, any> = Record
     schemaInvalidator = () => executorPlugin.invalidateUnifiedGraph();
   } else {
     let unifiedGraphFetcher: UnifiedGraphManagerOptions<unknown>['getUnifiedGraph'];
+
     if ('supergraph' in config) {
       unifiedGraphFetcher = () => handleUnifiedGraphConfig(config.supergraph, configContext);
-    } else if ('hive' in config) {
-      const fetcher = createSupergraphSDLFetcher(config.hive);
+    } else if (('hive' in config && config.hive.endpoint) || process.env.HIVE_CDN_ENDPOINT) {
+      const cdnEndpoint = 'hive' in config ? config.hive.endpoint : process.env.HIVE_CDN_ENDPOINT;
+      const cdnKey = 'hive' in config ? config.hive.key : process.env.HIVE_CDN_KEY;
+      if (!cdnKey) {
+        throw new Error(
+          'You must provide HIVE_CDN_KEY environment variables or `key` in the hive config',
+        );
+      }
+      const fetcher = createSupergraphSDLFetcher({
+        endpoint: cdnEndpoint,
+        key: cdnKey,
+      });
       unifiedGraphFetcher = () => fetcher().then(({ supergraphSdl }) => supergraphSdl);
+    } else {
+      // Falls back to `./supergraph.graphql` by default
+      unifiedGraphFetcher = () => {
+        try {
+          const res$ = handleUnifiedGraphConfig('./supergraph.graphql', configContext);
+          if ('catch' in res$ && typeof res$.catch === 'function') {
+            return res$.catch(e => {
+              if (e.code === 'ENOENT') {
+                throw new Error(
+                  'You must provide a supergraph schema in the `supergraph` config or `HIVE_CDN_ENDPOINT` environment variable or `./supergraph.graphql` file',
+                );
+              }
+              throw e;
+            });
+          }
+          return res$;
+        } catch (e) {
+          if (e.code === 'ENOENT') {
+            throw new Error(
+              'You must provide a supergraph schema in the `supergraph` config or `HIVE_CDN_ENDPOINT` environment variable or `./supergraph.graphql` file',
+            );
```

Comment:
Or point the config file to a supergraph schema.

Reply:
Done!